beaker-tests/README.md

```diff
@@ -0,0 +1,54 @@
+How to run tests
+================
+
+The following command will setup your system, setup a Fedora virtual machine and run tests from the `./tests/` subdirectory. You need to be root and the command will modify your system, i.e. install some packages, overwrite `/etc/rpkg/*.conf` configs and `/etc/rpkg.conf`, write an entry into `/etc/hosts`, and generates root ssh public key if not generated yet. It might be therefore a good idea to run this command inside another virtual machine (try with nested virtualization enabled) or a container:
+
+```
+# ./run.sh
```

Comment:
According to the description, I believe I should be able to do this?

    [jkadlcik@localhost dist-git]$ docker run -v /home/jkadlcik/git/dist-git:/distgit -it fedora:32 bash
    [root@1489a8739732 /]# cd /distgit/beaker-tests/
    [root@1489a8739732 beaker-tests]# ./run.sh

Which ends up with many errors for me. See full output here.
[dcc6c6a2.txt](https://github.com/release-engineering/dist-git/files/4638508/dcc6c6a2.txt)

Reply:
Yes, it doesn't even spawn the inner virtual machine because of some problem:

```
/usr/share/vagrant/gems/gems/vagrant-2.2.6/lib/vagrant/ui.rb:223: warning: The called method `say' is defined here
Error while connecting to libvirt: Error making a connection to libvirt URI qemu:///session?no_verify=1&keyfile=/root/.ssh/id_rsa:
Call to virConnectOpen failed: Failed to connect socket to '/var/run/libvirt/libvirt-sock': No such file or directory
```

I haven't really tested it because I run it directly on host.

Reply:
Probably, the problem is lack of systemd inside the container.
```
System has not been booted with systemd as init system (PID 1). Can't operate.
Failed to connect to bus: Host is down
```

Reply:
The question is what do we want to do with it? Because the paragraph above basically recommends running it inside of a virtual machine or container. Do we drop that recommendation, do we add another machine into `Vagrantfile` for running tests?

Reply:
Well, I think the recommendation is ok but we need perhaps we should be more specific, i.e. say "systemd-enabled container" instead of just container.

Reply:
The recommendation is good only as long as one knows how to achieve it. So I would say that it would be a really good idea to provide instructions how to do so.

Reply:
Well, nested virtualization is tricky. I just also found out that I need to change `/etc/libvirt/quemu.conf` records to contain:

```
user=root
group=root
```

for the vagrant machines to spawn. Otherwise I am getting errors like: `` I should probably add it to the README.

Where you already able to run the tests successfully?

I didn't actually manage to run it in container because of some network issues but it should be possible after. I would probably recommend using https://app.vagrantup.com/fedora/boxes/32-cloud-base for isolation. I am currently testing this.

> > > Yes, I don't think Tito is in epel-8 currently. In future, I would like to switch to rpkg-util for this project so I don't really mind this.
> > 
> > 
> > I thought that it already happened
> 
> Not yet.

Actually, you can build the dist-git with rpkg and I can see it is even recommended on the project page. But the doing it with auto-packing and rpkg-util v2 is a little bit cumbersome as you can see from tests. I think, right now both tools should be working for building but for new releases, tito is currently used.

Reply:
Actually, I was not able to make this work hitting at least three different issues (one being that my desktop cpu doesn't seem to support nested virtualization). I wouldn't therefor recommend neither solution and I will delete the recommendation. The correct way would be not to change system configs and instead rely on passing the path to testing configuration to the tools through command-line if that is possible. I would do it as an improvement, however.

Reply:
The problem is that the testing tooling like vagrant, vagrant-libvirt, beaker-libs etc. will still need to be installed but I don't see a better solution.

Reply:
>  I wouldn't therefor recommend neither solution and I will delete the recommendation. The correct way would be not to change system configs and instead rely on passing the path to testing configuration to the tools through command-line if that is possible. I would do it as an improvement, however.

+1, let's leave it for another PR. Thank you for the change.